### PR TITLE
[Feature] Support swap materialized view

### LINKED ISSUE #22561: Swap Materialized View
## Enhancement

Introduce a new syntax to update the definition of materialized view: 

```sql
CREATE MATERIALIZED VIEW mv_tmp AS xxxx;
ALTER MATERIALIZED VIEW mv1 SWAP WITH mv_tmp;
```

Behavior: 
1. The swap action is atomic, swap the name of two mvs

Use case:
1. Update schema of MV: add some columns in the materialized view
2. Change parameters of MV: change some parameters in materialized view

### PR DESCRIPTION
**Pull request by murphyatwork**
Fixes #22561

## What type of PR is this:
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 13 / 21 (61.90%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Table.java|0|3|00.00%|[669, 670, 672]|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|0|1|00.00%|[880]|
|:large_blue_circle:|com/starrocks/sql/ast/AlterMaterializedViewStmt.java|1|2|50.00%|[82]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|4|5|80.00%|[1649]|
|:large_blue_circle:|com/starrocks/alter/AlterJobMgr.java|8|10|80.00%|[343, 344]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 13 / 21 (61.90%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Table.java|0|3|00.00%|[669, 670, 672]|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|0|1|00.00%|[880]|
|:large_blue_circle:|com/starrocks/sql/ast/AlterMaterializedViewStmt.java|1|2|50.00%|[82]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|4|5|80.00%|[1649]|
|:large_blue_circle:|com/starrocks/alter/AlterJobMgr.java|8|10|80.00%|[343, 344]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 7 / 21 (33.33%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Table.java|0|3|00.00%|[669, 670, 672]|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|0|1|00.00%|[880]|
|:large_blue_circle:|com/starrocks/alter/AlterJobMgr.java|3|11|27.27%|[343, 344, 893, 894, 949, 950, 951, 952]|
|:large_blue_circle:|com/starrocks/sql/ast/AlterMaterializedViewStmt.java|1|2|50.00%|[82]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|3|4|75.00%|[1649]|

File: fe/fe-core/src/main/java/com/starrocks/alter/AlterJobMgr.java [hunk 1]
```diff
@@ -878,7 +880,7 @@ private void processSwap(Database db, OlapTable origTable, List<AlterClause> alt
         String origTblName = origTable.getName();
         String newTblName = clause.getTblName();
         Table newTbl = db.getTable(newTblName);
-        if (newTbl == null || !newTbl.isOlapOrCloudNativeTable()) {
+        if (newTbl == null || !(newTbl.isOlapOrCloudNativeTable() || newTbl.isMaterializedView())) {
```
Comment (LiShuMing) [inline fe/fe-core/src/main/java/com/starrocks/alter/AlterJobMgr.java:883]: How about `isCloudNativeMaterializedView`? we have too many table types...

File: fe/fe-core/src/main/java/com/starrocks/catalog/Table.java [hunk 2]
```diff
@@ -665,6 +665,12 @@ public Set<MvId> getRelatedMaterializedViews() {
         return relatedMaterializedViews;
     }
 
+    public void replaceRelatedMaterializedView(MvId oldId, MvId newId) {
```
Comment (LiShuMing) [inline fe/fe-core/src/main/java/com/starrocks/catalog/Table.java:668]: What's this used for?

File: fe/fe-core/src/main/java/com/starrocks/alter/AlterJobMgr.java (hunk 1 above)
Comment (mofeiatwork) [reply fe/fe-core/src/main/java/com/starrocks/alter/AlterJobMgr.java:883]: but that's different things:
1. `isMaterializedView`: `CLOUD_NATIVE_MATERIALIZED_VIEW`, `MATERIALIZED_VIEW`
2. `isOlapOrCloudNativeTable`: `OLAP`, `CLOUD_NATIVE`

and current code means, this item could either be a table, or an mv

File: fe/fe-core/src/main/java/com/starrocks/catalog/Table.java (hunk 2 above)
Comment (mofeiatwork) [reply fe/fe-core/src/main/java/com/starrocks/catalog/Table.java:668]: dead code. removed

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 7 / 18 (38.89%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|0|1|00.00%|[880]|
|:large_blue_circle:|com/starrocks/alter/AlterJobMgr.java|3|11|27.27%|[343, 344, 893, 894, 949, 950, 951, 952]|
|:large_blue_circle:|com/starrocks/sql/ast/AlterMaterializedViewStmt.java|1|2|50.00%|[82]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|3|4|75.00%|[1649]|

**Review by packy92 — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Pull request merged**